protocols/protocol/cloudstate/function.proto

```diff
@@ -45,17 +56,67 @@ message FunctionReply {
         Forward forward = 3;
     }
 
-    repeated SideEffect side_effects = 4;
+    repeated SideEffect side_effects = 5;
 }
 
 service StatelessFunction {
 
+    // Handle a unary command.
+    //
+    // The input command will contain the service name, command name, request metadata and the command
+    // payload. The reply may contain a direct reply, a forward or a failure, and it may contain many
+    // side effects.
     rpc handleUnary(FunctionCommand) returns (FunctionReply) {}
 
+    // Handle a streamed in command.
+    //
+    // The first message in will contain the request metadata, including the service name and command
```

Comment:
@jroper , I have been talking about this PR with @marcellanz and we have an important question, why do you say that metadata will be sent in the first message if the metadata is defined in the protocol and is contained within the message to be forwarded? I think the metadata in that case should be passed on message by message if the underlying protocols so support. Did you mean that metadata will not be forwarded by each message? That is, will subsequent messages not contain associated metadata?

Reply:
I'll clarify.

I think it should be this:

* If the underlying transport only supports per call metadata, that metadata will be on the first message, and not on the others.
* If the underlying transport supports per message metadata, each message will contain that metadata.

Reply:
Thanks @jroper 
We imagined it, thank you for clarifying that. It would be nice to add this text to the PR